src/Microsoft.Dism.Tests/ValidateProductKeyTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c). All rights reserved.
+//
+// Licensed under the MIT license.
+
+using Shouldly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Dism.Tests
+{
+    public class ValidateProductKeyTest : DismTestBase
+    {
+        public ValidateProductKeyTest(TestWimTemplate template, ITestOutputHelper testOutput)
+            : base(template, testOutput)
+        {
+        }
+
+        [Fact]
+        public void ValidateGenericProfessionalKey()
+        {
+            using (DismSession session = DismApi.OpenOnlineSession())
+            {
+                DismApi.ValidateProductKey(session, "VK7JG-NPHTM-C97JM-9MPGT-3V66T").ShouldBeTrue();
```

Comment:
On my machine, running Windows Enterprise, this test fails but it passes if I use a different product key: `XGVPP-NMH47-7TTHJ-W3FW7-8HV2C`

We might want to just omit this test for now or have fancy logic to detect the current edition and have a corresponding key.

Reply:
Tbh, I might have originally misunderstood the intend of that method... I will update the documentation a bit, and remove the test for now.

Regarding the _fancy logic_, I quite like the idea. I would do that in a different PR though, but something like a `WindowsEditionId` enum, and a collection of of `WindowsEdition` structs with default data that can be used during image servicing (i.e. name, description, generic key, etc.). What do you think?

Reply:
There's supposedly a way to read the current machine's product key, its stored in the registry as a `byte[]` and you have to decode it to a string but that might work best as the key to run the test against.  It just looks like a lot of code to decode the key.

https://github.com/mrpeardotnet/WinProdKeyFinder/blob/master/WinProdKeyFind/KeyDecoder.cs#L115

Reply:
I don't mind the code, the issue is that with digital licenses (e.g. Microsoft account), the _key_ part of the byte array is zero'ed out. This is the case on my machine for example.

I kept digging: `pkeyhelper.dll` exports the function `SkuGetProductKeyForEdition` which returns a default key for a given edition id. As far as I can tell, the entire code base behind it is exactly the same as what `TransmogProvider` uses internally, same function names, signatures and DBs. (The provider uses it when you call SetEdition without specifying a product key.)

I'm not sure how useful `DismApi.GetDefaultProductKey(DismSession session)` would be. What do you think? At the very least I can fix the test.

Reply:
Since we've both tested manually that the method works and I doubt there's any chance of regressions since these APIs are very locked in, I'm not sure if tests really matter.  But if anything, you could add the logic to call `SkuGetProductKeyForEdition()` and pass that product key in the test.  I would not bother exposing it in `DismApi` since its not really part of DISM.

Reply:
I tend to agree. I'll keep a reference to what I've got for a potential future use, but let's leave it as is for now.